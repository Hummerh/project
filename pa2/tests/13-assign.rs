fn main() {
      let mut x = 1;
      let y = 2;

      x = x + y;
      x += y;
      x -= y * 2;
      x *= y;
      x /= y / y;
      x %= y;
}
