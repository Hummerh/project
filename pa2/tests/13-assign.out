(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-mut-id
                        (id
                            (x)))
                    (lit-dec))
                (let
                    (pat-id
                        (id
                            (y)))
                    (lit-dec))
                (assign
                    (id
                        (x))
                    (add
                        (id
                            (x))
                        (id
                            (y))))
                (assign-add
                    (id
                        (x))
                    (id
                        (y)))
                (assign-sub
                    (id
                        (x))
                    (mul
                        (id
                            (y))
                        (lit-dec)))
                (assign-mul
                    (id
                        (x))
                    (id
                        (y)))
                (assign-div
                    (id
                        (x))
                    (div
                        (id
                            (y))
                        (id
                            (y))))
                (assign-rem
                    (id
                        (x))
                    (id
                        (y)))))))